### ISSUE TITLE
Sik-Ho Tang | Review: Trimps-Soushen -- Winner in ILSVRC 2016 (Image Classification).

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#106]
Sik-Ho Tang. [Review: Trimps-Soushen — Winner in ILSVRC 2016 (Image Classification)](https://towardsdatascience.com/review-trimps-soushen-winner-in-ilsvrc-2016-image-classification-dfbc423111dd).

**Comment by NorbertZheng** [comment]
## Overview
Inthis story, the approach by the winner, Trimps-Soushen, in ILSVRC 2016 classification task, is reviewed. Trimps stands for The Third Research Institute of Ministry of Public Security, or in chinese 公安部三所. In brief, Trimps is the research institute for advancing the technologies for public security in China, which was launched in 1978 at Shanghai. Soushen should be the team name under Trimps, in chinese 搜神. It means god of search, where Sou (搜) means search and Shen (神) means god.

Trimps-Soushen has won several competitions in 2016:
 - Object Localization: 1st place, 7.71% error.
 - Object Classification: 1st place, 2.99% error.
 - Object Detection: 3rd place, 61.82% mAP.
 - Scene Classification: 3rd place, 10.3% error.
 - Object Detection from video: 3rd place, 70.97% mAP.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/ed48b32b-6aab-402f-83e8-1afd055d9d3c)
ILSVRC 2016 Classification Ranking http://image-net.org/challenges/LSVRC/2016/results#loc

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/1b5a9ead-ca2a-42d1-88b1-4ec1e8b48919)
ILSVRC Classification Results from 2011 to 2016.

Though Trimps-Soushen has the state-of-the-art results on multiple recognition tasks, there is no new innovative technology or novelty by Trimps-Soushen. Maybe due to this reason, they haven’t published any papers or technical reports about it.

Instead, they only shared their results in the ImageNet and COCO joint workshop in 2016 ECCV. And they have some funny facts about the dataset.

**Comment by NorbertZheng** [comment]
## Ensemble Using Different Models
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/096c0e68-c99d-434c-9490-e591b9a42aa9)
ImageNet Classification Errors for Top-10 Difficult Categories.

Trimps-Soushen used the pretrained models from [Inception-v3](https://medium.com/@sh.tsang/review-inception-v3-1st-runner-up-image-classification-in-ilsvrc-2015-17915421f77c), [Inception-v4](https://towardsdatascience.com/review-inception-v4-evolved-from-googlenet-merged-with-resnet-idea-image-classification-5e8c339d18bc), [Inception-ResNet-v2](https://towardsdatascience.com/review-inception-v4-evolved-from-googlenet-merged-with-resnet-idea-image-classification-5e8c339d18bc), [Pre-Activation ResNet-200](https://towardsdatascience.com/resnet-with-identity-mapping-over-1000-layers-reached-image-classification-bb50a42af03e), and [Wide ResNet (WRN-68–2)](https://towardsdatascience.com/review-wrns-wide-residual-networks-image-classification-d3feb3fb2004) for classification, and found out Top-10 difficult categories as above.
 - Diverse results are obtained, which means there is no models being dominant for all categories. **Each of the models are strong at classifying some categories, but also weak at classifying some categories.**
 - The diversity of models can be used for improving the accuracy. E.g. Boosting!!!

During training, Trimps-Soushen just performed multi-scale augmentation & large mini batch size. During testing, multi-scale + flip are used with dense fusion.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/1be90063-c945-491c-9c11-1c69cb66a4eb)
ImageNet Top-5 Error Rate Results.

 - The validation errors for the 5 best models are from 3.52% to 4.65%.
 - By ensembling these 5 models ([Inception-ResNet-v2](https://towardsdatascience.com/review-inception-v4-evolved-from-googlenet-merged-with-resnet-idea-image-classification-5e8c339d18bc) has higher weight), 2.92% validation error for is obtained.
 - And 2.99% test error is obtained which is the first to obtain under 3% error rate.

**Comment by NorbertZheng** [comment]
## Some Findings Based on Top-20 Accuracy
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/800cba1a-e626-4092-8b8d-fa5817f3550e)
Top-k Accuracy.

Top-k Accuracy is obtained as shown above. When k=20, 99.27% accuracy is obtained. The error rate is smaller than 1%.

 - Why there are still errors when Top-20 accuracy is used?

**Trimp-Soushen has analysed those 1% error images in very detail!!!**

They manually analysed 1458 error images from validation set. And roughly 7 categories of errors are obtained as below:

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/3e7b18cd-ead3-4029-9cb0-ec13686f9dd1)
7 Error Categories.

**Comment by NorbertZheng** [comment]
### Label May Wrong
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/816358ee-c51e-47cc-9011-84aa98c34a80)
Label May Wrong (Maybe it is really a sleeping bag for Hello Kitty? lol).

The ground truth is sleeping bag, but obviously, it is a pencil box!!!!

This is because the ground truths are manually labelled in ImageNet dataset. As ImageNet is a dataset of over 15 millions labeled high-resolution images with around 22,000 categories, and a subset of 1000-categories ImageNet dataset is used for competition, there would be some wrong labels.

There are 211 out of 1458 error images which are “labels May Wrong”, which is about 15.16%.

**Comment by NorbertZheng** [comment]
### Multiple Objects (>5)
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/25c88035-9ae8-4391-abb1-a4729cde12d3)
Multiple Objects (>5) (Which is the main object?).

The above image contains multiple objects (>5). Actually this kind of images is not suitable for ILSVRC classification task. Because in ILSVRC classification task, only one class should be identified for each image.

There are 118 out of 1458 error images which are “Multiple Objects (>5)”, which is about 8.09%.

**Comment by NorbertZheng** [comment]
### Non-Obvious Main Object
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/16da7527-d870-4fe3-a930-43e7a7a6ca1a)
Non-Obvious Main Object (Please find the paper towel in the image, lol !!)

As only one class should be identified in classification task, the above image does not have one obvious main object in the image. It can be boat or dock. But the ground truth is paper towel.

There are 355 out of 1458 error images which are “Non-Obvious Main Object”, which is about 24.35%.

**Comment by NorbertZheng** [comment]
### Confusing Label
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/fe40ea3f-5310-44df-a6c3-1d4ea0e7f218)
Confusing Label (Maybe there is no sunscreen inside, lol.)

The ground truth is sunscreen. This time, the label seems to be correct as the carton saying about SPF30. But the task would become to understand the meaning of the text on the carton, which is going too far from the original objective of recognizing objects based on shapes and colors.

There are 206 out of 1458 error images which are “Confusing Label”, which is about 14.13%.

 - Require semantic understanding!!!

**Comment by NorbertZheng** [comment]
### Fine-Grained Label
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/f625afef-3db5-4800-91cd-b636b88beac5)
Fine-Grained Label.

The ground truth is correct. Both bolete and stinkhorn are the types of fungal. Indeed, this type of label is even difficult for human to identify.

There are 258 out of 1458 error images which are “Fine-Grained Label”, which is about 17.70%.

The network can improve this category.

**Comment by NorbertZheng** [comment]
### Obvious Wrong
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/97ea79d3-9e5c-4ba4-b5fd-4ac4866e0a3c)
Obvious Wrong.

The ground truth is correct. And the network cannot predict it even using top-20 prediction.

There are 234 out of 1458 error images which are “Obvious Wrong”, which is about 16.05%.

The network can improve this category.

**Comment by NorbertZheng** [comment]
### Partial Object
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/63f4d85b-104f-4ada-b5df-2fe3b6eb2e6e)
Partial Object
The image may only contain a part of the object, which is hard to recognize. Maybe the image can be better if it is zoomed out with multiple tables and chairs, looking like a restaurant.

There are 66 out of 1458 error images which are “Partial Object”, which is about 4.53%.

**Comment by NorbertZheng** [comment]
Therefore, the accuracy is hard to be improved by 1%.

**Comment by NorbertZheng** [comment]
## Region Fusion (Image Localization)
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/4e0b4143-31f0-4c6b-9166-3e8de6aed740)
Region Fusion for Image Localization.

To localize the top-5 predicted labels within the image, [Faster R-CNN](https://towardsdatascience.com/review-faster-r-cnn-object-detection-f5685cb30202) architecture using multiple models is used. Multiple models are used to generate region proposals via the region proposal network (RPN) in [Faster R-CNN](https://towardsdatascience.com/review-faster-r-cnn-object-detection-f5685cb30202). Then based on the top-5 classification predicted labels, to perform the localization prediction.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/52abc891-4758-4044-94e6-a2bb2aa8287a)
Image Localization Top-5 Validation Error Results.

 - Previously state-of-the-art approaches: 8.51% to 9.27% error are obtained.
 - Ensemble all: By fusion all approaches together, 7.58% error is achieved.
 - Ensemble all but without one model: Only 7.75 to 7.93% error is obtained.

Thus, diversity between model is important and contribute large improvement on prediction accuracy.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/708bdfd5-f19a-4e93-9e8e-76dfb8efc2bd)
ILSVRC Localization Top-5 Test Error Results from 2012 to 2016.

**Comment by NorbertZheng** [comment]
## Multi-Model Fusion for Other Tasks (Scene Classification / Object Detection / Object Detection from Video)
### Scene Classification
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/7e9a29d6-3d8c-4798-abe2-ed496627fb28)
Multi-Scale & Multi-Model Fusion for Scene Classification.

Improved multi-scale approach is used by concatenating the results instead of just adding the results together at the end of network.

On top of using multi-scale inputs for prediction using the same model, two trained models (I believe using the same model network) are used by concatenating the results and goes through FC and softmax. 10.80% validation error is obtained.

By using 7 * two models, 10.39% validation error and 10.42% test error are obtained.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/afaa76ec-7030-4ffe-aad7-3136f634ef94)
Scene Classification Top-5 Test Error Results.

With the model using external dataset, Places2, for pretraining as well, 10.3% top-5 test error is obtained which got 3rd place in scene classification.

**Comment by NorbertZheng** [comment]
### Object Detection
Similar to image localization, [Faster R-CNN](https://towardsdatascience.com/review-faster-r-cnn-object-detection-f5685cb30202) architecture is used with multi-model fusion.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/dcc6c4fc-0e8b-4a6a-8214-ffdf74898d32)
Object Detection mAP Results.

61.82% mAP is obtained.

**Comment by NorbertZheng** [comment]
### Object Detection from Video
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a4d2ab1c-902f-4d31-bff5-0db912299fdd)
Object Detection from Video mAP Results.

Optical flow guided motion prediction is also used to reduce false negative detection. 70.97% mAP is obtained.

**Comment by NorbertZheng** [comment]
Due to the model diversity, model fusion is effective. By using model fusion, Trimps-Soushen outperforms [ResNeXt](https://towardsdatascience.com/review-resnext-1st-runner-up-of-ilsvrc-2016-image-classification-15d7f17b42ac) and [PolyNet](https://towardsdatascience.com/review-polynet-2nd-runner-up-in-ilsvrc-2016-image-classification-8a1a941ce9ea) and got 1st place in image classification. And model fusion is also successfully applied to other tasks.
 - This means that except the innovative optimization or novel design of network architecture, other technical stuffs such as multi-model fusion can also help to improve the accuracy a lot.

On the other hand, if Trimps-Soushen uses [ResNeXt](https://towardsdatascience.com/review-resnext-1st-runner-up-of-ilsvrc-2016-image-classification-15d7f17b42ac) and [PolyNet](https://towardsdatascience.com/review-polynet-2nd-runner-up-in-ilsvrc-2016-image-classification-8a1a941ce9ea) for model fusion, perhaps the error could be further reduced, since [ResNeXt](https://towardsdatascience.com/review-resnext-1st-runner-up-of-ilsvrc-2016-image-classification-15d7f17b42ac) and [PolyNet](https://towardsdatascience.com/review-polynet-2nd-runner-up-in-ilsvrc-2016-image-classification-8a1a941ce9ea) obtain higher accuracy compared with those models used for model fusion.

**Comment by NorbertZheng** [comment]
## References
 - [2016 ECCV] [Trimps-Soushen] (Slides Only) [Good Practices for Deep Feature Fusion](http://image-net.org/challenges/talks/2016/Trimps-Soushen@ILSVRC2016.pdf).
